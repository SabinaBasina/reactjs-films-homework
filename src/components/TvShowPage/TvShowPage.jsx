import React, { Component } from 'react';
// import http from 'axios';
import PropTypes from 'prop-types';
import TvShow from '../TvShow/TvShow';
import Search from '../Search';
import styles from './TvShowPage.scss';
import Loading from './Loading.gif';

class TvShowPage extends Component {
  state = {
    isReady: false,
  };


  componentDidMount() {
    const { loadTvShows } = this.props;
    loadTvShows();
    this.setReadyState(true);
  }

  componentDidUpdate(prevProps) {
    const { page, loadTvShows } = this.props;
    const { isReady } = this.state;
    if ((page !== prevProps.page) && isReady) {
      loadTvShows();
    }
  }

  setReadyState = (newIsReady) => {
    const { onReady } = this.props;
    this.setState({ isReady: newIsReady });
    onReady(newIsReady);
  }

  render() {
    // const { isReady } = this.state;
    const { tvShows } = this.props;

    if (!tvShows) {
      return null;
    }
    return (

      <main className={styles.TvShows}>

        {!tvShows
          && (
            <img
              src={Loading}
              className={styles.Ready}
              alt="Loading page"
            />
          )}

        {tvShows
          && (
            <div>
              <Search />
              <div className={styles.TvShowLibrary}>
                {tvShows.map(tvShowData => (
                  <TvShow data={tvShowData} />
                ))}
              </div>
            </div>
          )}

      </main>

    );
  }
}

export default TvShowPage;

TvShowPage.propTypes = {
  loadTvShows: PropTypes.func,
  onReady: PropTypes.func,
  page: PropTypes.number,
  tvShows: PropTypes.arrayOf(PropTypes.array).isRequired,
};

TvShowPage.defaultProps = {
  loadTvShows: () => { },
  onReady: () => { },
  page: 0,
};
