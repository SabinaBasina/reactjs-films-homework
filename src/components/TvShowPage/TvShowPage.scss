.TvShows {
  text-align: center;
}

.TvShowLibrary {
  margin-left: 15%;
  margin-right: 15%;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(260px, 1fr));
}
