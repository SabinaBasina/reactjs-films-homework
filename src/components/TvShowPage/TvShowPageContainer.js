import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';

import TvShowPage from './TvShowPage';
import { loadTvShows } from '../../moduleActions/tvshowpage';
import { getTvShows } from '../../moduleSelectors';

const mapStateToProps = createStructuredSelector({
  tvShows: getTvShows,
});

const mapDispatchToProps = {
  loadTvShows,
};

export default connect(mapStateToProps, mapDispatchToProps)(TvShowPage);
