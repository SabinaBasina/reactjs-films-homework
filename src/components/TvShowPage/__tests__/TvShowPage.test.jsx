import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TvShowPage from '../TvShowPage';


test('TvShowPage test', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<TvShowPage />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
