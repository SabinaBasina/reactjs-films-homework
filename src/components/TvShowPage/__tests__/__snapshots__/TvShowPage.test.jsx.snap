// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TvShowPage test 1 1`] = `
<main
  className="TvShows"
>
  <div>
    <withRouter(Search) />
    <div
      className="TvShowLibrary"
    >
      <TvShow
        data={
          Object {
            "id": 1,
          }
        }
      />
      <TvShow
        data={
          Object {
            "id": 2,
          }
        }
      />
    </div>
  </div>
</main>
`;

exports[`test TvShowPage test 2 1`] = `null`;

exports[`test TvShowPage test 3 1`] = `null`;
